src/DatePicker/DatePicker.js

```diff
@@ -277,22 +360,36 @@ class DatePicker extends Component {
       shouldDisableDate,
       style,
       textFieldStyle,
+      keyboardEnabled, // eslint-disable-line no-unused-vars
+      errorText,
       ...other
     } = this.props;
 
     const {prepareStyles} = this.context.muiTheme;
-    const formatDate = formatDateProp || this.formatDate;
+    const styles = getStyles(this.props, this.context);
 
     return (
       <div className={className} style={prepareStyles(Object.assign({}, style))}>
         <TextField
           {...other}
+          onBlur={this.handleBlur}
+          onChange={this.handleTextChange}
           onFocus={this.handleFocus}
-          onTouchTap={this.handleTouchTap}
+          onTouchTap={!this.shouldHandleKeyboard() && this.handleTouchTap}
           ref="input"
           style={textFieldStyle}
-          value={this.state.date ? formatDate(this.state.date) : ''}
+          errorText={this.state.hasError && errorText}
```

Comment:
That's a breaking change as users might be using this `errorText` property independently from the `hasError` state. I would rather move that error logic away into another state error wrapper component.